conversant/prompt_chatbot.py

```diff
@@ -244,79 +257,127 @@ def generate_prompt_update_examples(self, query: str) -> str:
             )
         return current_prompt
 
-    def append_to_chat_history(
```

Comment:
Judgment call I'm making: there is no need to factor out these lines of code into its own function as it is only called twice in the `partial_reply` function, one with the flag on and another off.